apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/instructorAiGenerateDrafts.ts

```diff
@@ -29,4 +32,41 @@ router.get(
   }),
 );
 
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_permission_edit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be course editor)');
+    }
+    if (req.body.__action === 'delete_drafts') {
+      const questions = await queryRows(
+        sql.select_drafts_by_course_id,
+        { course_id: res.locals.course.id.toString() },
+        QuestionSchema,
+      );
+
+      const client = getCourseFilesClient();
+
+      const result = await client.batchDeleteQuestions.mutate({
+        course_id: res.locals.course.id,
+        user_id: res.locals.user.user_id,
+        authn_user_id: res.locals.authn_user.user_id,
+        has_course_permission_edit: res.locals.authz_data.has_course_permission_edit,
+        question_ids: questions.map((question) => question.id),
+      });
+
+      if (result.status === 'error') {
+        throw new error.HttpStatusError(
+          500,
+          `Cannot delete all draft questions in: (${questions.map((question) => question.qid).join(', ')})`,
+        );
+      }
+
+      res.redirect(`${res.locals.urlPrefix}/ai_generate_question_drafts`);
```

Comment:
Can you just redirect to `req.originalUrl` instead?

Reply:
Done in fd0f052ad2ce1e73b49a4680c35f09c149548574.